MiniSpace.Services.Reactions/src/MiniSpace.Services.Reactions.Infrastructure/Mongo/Queries/Handlers/GetReactionsSummaryHandler.cs

```diff
@@ -0,0 +1,46 @@
+using Convey.CQRS.Queries;
+using Convey.Persistence.MongoDB;
+using MiniSpace.Services.Reactions.Application.Dto;
+using MiniSpace.Services.Reactions.Application.Queries;
+using MiniSpace.Services.Reactions.Core.Entities;
+using MiniSpace.Services.Reactions.Infrastructure.Mongo.Documents;
+using MongoDB.Driver;
+using MongoDB.Driver.Linq;
+
+namespace MiniSpace.Services.Reactions.Infrastructure.Mongo.Queries.Handlers
+{
+    public class GetReactionsSummaryHandler : IQueryHandler<GetReactionsSummary, ReactionsSummaryDto>
+    {
+        private readonly IMongoRepository<ReactionDocument, Guid> _reactionRepository;
+
+        public GetReactionsSummaryHandler(IMongoRepository<ReactionDocument, Guid> reactionRepository)
+        {
+            _reactionRepository = reactionRepository;
+        }
+        
+        public async Task<ReactionsSummaryDto>
+            HandleAsync(GetReactionsSummary query, CancellationToken cancellationToken)
+        {
+            var documents = _reactionRepository.Collection.AsQueryable();
+            documents = documents.Where(p => p.ContentId == query.ContentId && p.ContentType == query.ContentType);
+
+            var reactions = await documents.ToListAsync();
+
+            // Get number for each reaction type
+            Dictionary<ReactionType, int> nrRcs = new();
+            foreach (var r in reactions) {
+                int tmp = 0;
+                nrRcs.TryGetValue(r.Type, out tmp);
+                nrRcs[r.Type] = tmp + 1;
```

Comment:
I am not sure what is going on here. It would be easier to just use GroupBy Select and OrderBy on documents.

Reply:
Also I would like to have reaction of authenticated user in ReactionsSummaryDto. I think the other endpoint GetReactions should be used only if you want to check who gave what reaction. So there is some sort of information if user gave the reaction or not. If the user is not authenticated or didn't give reaction yet, the reaction should be null.

Reply:
It would be nice to also pass reaction type not just its Id